src/test/java/io/odh/test/unit/MultipartFormDataBodyPublisherTests.java

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright Skodjob authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+// Copyright (c) 2022 yusuke suzuki
+package io.odh.test.unit;
+
+import com.sun.net.httpserver.HttpServer;
+import io.odh.test.TestSuite;
+import io.odh.test.platform.httpClient.MultipartFormDataBodyPublisher;
+import lombok.SneakyThrows;
+import org.hamcrest.Matchers;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.InetSocketAddress;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse.BodyHandlers;
+import java.nio.ByteBuffer;
+import java.nio.channels.Channels;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.Flow;
+import java.util.regex.Pattern;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+
+@Tag(TestSuite.UNIT)
+public class MultipartFormDataBodyPublisherTests {
```

Comment:
it should also implement `TestVisualSeparator` and extend `@ExtendWith(ExtensionContextParameterResolver.class)` like unittest class